core/card.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Drone IO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"io"
+
+	"github.com/drone/drone/handler/api/errors"
+)
+
+var (
+	errCardStepInvalid   = errors.New("No Step ID Provided")
+	errCardBuildInvalid  = errors.New("No Build ID Provided")
+	errCardSchemaInvalid = errors.New("No Card Schema Has Been Provided")
+	errCardDataInvalid   = errors.New("No Card Data Has Been Provided")
+)
+
+type Card struct {
+	Id     int64  `json:"id,omitempty"`
+	Build  int64  `json:"build,omitempty"`
+	Stage  int64  `json:"stage,omitempty"`
+	Step   int64  `json:"step,omitempty"`
+	Schema string `json:"schema,omitempty"`
+}
+
+type CreateCard struct {
+	Id     int64  `json:"id,omitempty"`
+	Build  int64  `json:"build,omitempty"`
+	Stage  int64  `json:"stage,omitempty"`
+	Step   int64  `json:"step,omitempty"`
+	Schema string `json:"schema,omitempty"`
+	Data   []byte `json:"data,omitempty"`
+}
```

Comment:
I think we should be able to eliminate the `CreateCard` struct since the `Data` should be passed to `CreateCard` as an `io.Reader`.  

```
CreateCard(ctx context.Context, card *Card, data io.Reader) error
```

Reply:
done,